server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java

```diff
@@ -129,12 +139,31 @@ protected EmptyResult shardOperation(ClearIndicesCacheRequest request, ShardRout
             shardRouting.shardId(),
             request.queryCache(),
             request.fieldDataCache(),
-            request.requestCache(),
+            false,
             request.fields()
         );
+        // For request cache clearing, only accumulate CacheEntity for each shard, and save marking and clearing for the nodeOperation.
+        // Then the scheduled cache clear won't happen midway through.
+        ShardId shardId = shardRouting.shardId();
+        IndexShard shard = indicesService.indexService(shardId.getIndex()).getShardOrNull(shardId.id());
+
+        if ((request.requestCache() || ((request.queryCache() || request.fieldDataCache()) == false && request.fields().length == 0))
+            && shard != null) {
+            requestCacheEntities.add(new IndicesService.IndexShardCacheEntity(shard));
```

Comment:
I don't think we need to pass `IndexShardCacheEntity` necessarily here, just pass shardId and let indicesService take care of creating a IndexShardCacheEntity. That way we don't need to make this class public.